src/compiler/parser.ts

```diff
@@ -2733,7 +2733,7 @@ namespace ts {
                 entity = finishNode(
                     factory.createQualifiedName(
                         entity,
-                        parseRightSideOfDot(allowReservedWords, /* allowPrivateIdentifiers */ false) as Identifier
+                        parseRightSideOfDot(allowReservedWords, allowPrivateIdentifiers) as Identifier

```

Comment:
What is the shape of the AST here?
QualifiedName expects an `Identifier` as the right node.
But instead it receives `PrivateIdentifier`.